packages/component-library/src/components/form/mt-text-editor/_internal/mt-text-editor-toolbar-button-link.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <mt-text-editor-toolbar-button
```

Comment:
When this button is the only thing that's triggering the modal you can use the `<mt-modal-trigger :as="MtTextEditorToolbarButton">` component. That way you don't need to use the `isOpen` property

You can take a look at the `mt-modal.spec.ts` file to see how that works.

Reply:
This is cool. I think we need some documentation for the new mt-modal component and how it is working 😄

Reply:
It seems like that it only opens it and I can't do anything besides this (like setting the attributes from the editor, etc.). So it won't work for me.

```
const openLinkModal = () => {
  // Get current link from selection
  linkHref.value = props.editor.getAttributes("link").href ?? "";
  linkTarget.value = props.editor.getAttributes("link").target ?? "";

  showLinkModal.value = true;
};
```

Reply:
Ok, if that's not working for you then it's fine to do it like this.